test/uk/gov/hmrc/traderservices/journey/AmendCaseJourneyModelSpec.scala

```diff
@@ -1075,30 +1189,87 @@ class AmendCaseJourneyModelSpec extends UnitSpec with StateMatchers[State] with
       }
 
       "goto AmendCaseConfirmation when amendCase" in {
+        val updateCaseApi: UpdateCaseApi = { request =>
+          Future.successful(TraderServicesCaseResponse(correlationId = "", result = Some("PC12010081330XGBNZJO04")))
+        }
         given(
           FileUploaded(
             fullAmendCaseStateModel,
-            FileUploads(files =
-              Seq(
-                FileUpload.Accepted(
-                  1,
-                  "foo-bar-ref-1",
-                  "https://bucketName.s3.eu-west-2.amazonaws.com?1235676",
-                  ZonedDateTime.parse("2018-04-24T09:30:00Z"),
-                  "396f101dd52e8b2ace0dcf5ed09b1d1f030e608938510ce46e7a5c7a4e775100",
-                  "test.pdf",
-                  "application/pdf"
-                )
-              )
-            ),
+            someFileUploads,
             acknowledged = false
           )
         ) when
-          amendCase(eoriNumber) should
+          amendCase(updateCaseApi)(eoriNumber) should
           thenGo(
             AmendCaseConfirmation("PC12010081330XGBNZJO04")
           )
       }
+
+      "fail when amendCase and received error response" in {
+        val updateCaseApi: UpdateCaseApi = { request =>
+          Future.successful(
+            TraderServicesCaseResponse(correlationId = "", error = Some(ApiError("", Some(""))))
+          )
+        }
+        given(
+          FileUploaded(
+            fullAmendCaseStateModel,
+            someFileUploads,
+            acknowledged = false
+          )
+        ) shouldFailWhen amendCase(updateCaseApi)(eoriNumber)
+      }
+
+      "fail when amendCase and received case reference doesn't match" in {
+        val updateCaseApi: UpdateCaseApi = { request =>
+          Future.successful(
+            TraderServicesCaseResponse(correlationId = "", result = Some("not_the_same_case_reference"))
+          )
+        }
+        given(
+          FileUploaded(
+            fullAmendCaseStateModel,
+            someFileUploads,
+            acknowledged = false
+          )
+        ) shouldFailWhen amendCase(updateCaseApi)(eoriNumber)
+      }
+
+      "goto EnterCaseReferenceNumber when amendCase without case reference number provided" in {
+        val updateCaseApi: UpdateCaseApi = { request =>
+          Future.successful(TraderServicesCaseResponse(correlationId = "", result = Some("PC12010081330XGBNZJO04")))
+        }
+        val model = fullAmendCaseStateModel.copy(caseReferenceNumber = None)
+        given(
+          FileUploaded(
+            model,
+            someFileUploads,
+            acknowledged = false
+          )
+        ) when
+          amendCase(updateCaseApi)(eoriNumber) should
+          thenGo(
+            EnterCaseReferenceNumber(model.copy(fileUploads = Some(someFileUploads)))
+          )
+      }
+
+      "goto SelectTypeOfAmendment when amendCase without case reference number provided" in {
```

Comment:
V minor but I think this test name should be 'without type of amendment' or something along those lines?